xee/ext.py

```diff
@@ -555,11 +559,19 @@ def get_variables(self) -> utils.Frozen[str, xarray.Variable]:
     lon_grid = self.project((0, 0, v0.shape[1], 1))
     lat_grid = self.project((0, 0, 1, v0.shape[2]))
     lon = self.image_to_array(
-        lnglat_img, grid=lon_grid, dtype=np.float32, bandIds=['longitude']
+        lnglat_img,
+        grid=lon_grid,
+        dtype=np.float32,
+        bandIds=['longitude', 'latitude'],
```

Comment:
Yes, `pyproj.Transformer.transform` requires both the x and y coordinates be given.

If my understanding of `lon_grid` and `lat_grid` is correct, then the original calls to `self.image_to_array` return the longitude of the topmost and latitude of the leftmost edges of the bounds respectively. If we give those as the x and y inputs to the transformer, what we get back is the transformed coordinates of the diagonal of the bounds which differs from transforming the coordinates of the leftmost and topmost edges separately which is what I'm doing and why latitude is included in the `image_to_array` call to get the longitude and vice versa.

Looking at this again, I  think that using multidimensional coordinates may be a better solution because the difference between the two methods occurs because the x coordinates vary along the y axis and the y coordinates vary along the x axis...

Open to input on that though.

[Here is a notebook](https://colab.research.google.com/drive/1XliUoBkpFUtxsxDwOocjYZIvDiArPoLp?usp=sharing) that shows the difference

Reply:
I went back and looked at this again and I believe the way I'm doing it is correct. 

When I open a dataset -> grab the first image -> write to a geotiff using rioxarray -> upload back to earthengine -> compare against the original first image in the collection I get an exact match. See this [notebook](https://colab.research.google.com/drive/1zs20Vqcjlvx5mxdapENOfdgu-BH1itgq?usp=sharing)

Additionally, extracting the top edge and left edge without grabbing both the longitude and latitude and transforming those together will fail if the region is not perfectly square because `pyproj.Transformer.transform` needs the x inputs and y inputs to have the same length.